pkg/ext/delegate_error.go

```diff
@@ -0,0 +1,110 @@
+package ext
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+// delegateError wraps an inner delegate and converts unknown errors.
+type delegateError[T runtime.Object, TList runtime.Object] struct {
+	inner *delegate[T, TList]
+}
+
+func (d *delegateError[T, TList]) convertError(err error) error {
+	if _, ok := err.(errors.APIStatus); ok {
+		return err
+	}
+
+	return errors.NewInternalError(err)
+}
+
```

Comment:
This can be a function on its own as it doesn't use any fields/methods of the inner delegate. It would also be simpler to test (since you don't need to construct such a inner delegate).

Reply:
Agreed. Thanks!